src/pages/api/gallery.ts

```diff
@@ -0,0 +1,49 @@
+import type { APIRoute } from "astro";
+import {
+	GameMetadata,
+	getGalleryGames,
+} from "../../lib/game-saving/gallery";
+
+export const get: APIRoute = async ({ request }) => {
+	const params = new URL(request.url).searchParams;
+
+	let games = getGalleryGames().map((game) => ({
+		...game,
+		title: game.title.trim(),
+		lowerCaseTitle: game.title.trim().toLowerCase(),
+		lowerCaseAuthor: game.author.toLowerCase(),
+	}));
+
+	games = games
+		.sort((a, b) => Date.parse(b.addedOn) - Date.parse(a.addedOn))
+		.slice(0, 10)
+		.map((game) => ({ ...game, isNew: true } as GameMetadata))
+		.concat(games.slice(10));
+
+	if (params.has("new")) {
+		games = games.filter((game) => game.isNew);
+	}
+
+	if (params.has("query")) {
+		const query = params.get("query")?.toLowerCase() || "";
```

Comment:
if `query` is in the params then we don't need to use optional chaining right? We can use `params.get('query')!` (the bang) to tell TypeScript it's not going to be null